src/main/java/io/codelex/knowledgecheck/exercise1/CreditCard.java

```diff
@@ -0,0 +1,25 @@
+package io.codelex.knowledgecheck.exercise1;
+
+public class CreditCard extends Card {
+
+    public CreditCard(String number, String owner, String ccv, double balance) {
+        super(number, owner, ccv, balance);
+    }
+
+    @Override
+    public void addMoney(double amount) {
+        balance += amount;
+    }
+
+    @Override
+    public void takeMoney(double amount) throws NotEnoughFundsException {
+        if (amount <= balance) {
+            balance -= amount;
+            if (balance < 100) {
```

Comment:
Šo IF varētu pārnest vispār šis metodes beigās:
```
        if (amount <= balance) {
            balance -= amount;
        } else {
            throw new NotEnoughFundsException("Not enough funds!");
        }
       if (balance < 100) {
            System.out.println("Warning: Low funds");
       }
```
<= tad varētu pamanīt, ka metodes pirmā puse sakrīt ar DebitCard metodes takeMoney pirmo pusi. Un tad varētu secināt, ka šādu metodi varētu likt abstraktajā Card klasē:
```
    public void takeMoney(double amount) throws NotEnoughFundsException {
        if (amount <= balance) {
            balance -= amount;
        } else {
            throw new NotEnoughFundsException("Not enough funds!");
        }
    }
```
Un šeit atstāt tikai :
```
super.takeMoney(amount)
if (balance < 100) {
    System.out.println("Warning: Low funds");
}
```